js/utils/keyboardNav.js

```diff
@@ -0,0 +1,292 @@
+        }
+
+        switch (event.keyCode) {
+            case this.KEYCODE_RIGHT:
+                this.navigateNextBlock();
+                event.preventDefault();
+                break;
+                
+            case this.KEYCODE_LEFT:
+                this.navigatePrevBlock();
+                event.preventDefault();
+                break;
+                
+            case this.KEYCODE_UP:
+                this.navigatePrevBlock();
+                event.preventDefault();
+                break;
+                
+            case this.KEYCODE_DOWN:
+                this.navigateNextBlock();
+                event.preventDefault();
+                break;
+                
+            case this.KEYCODE_ESC:
+                this.toggleKeyboardNavigation();
+                event.preventDefault();
+                break;
+        }
+    }
+
+    /**
+     * Navigate to the next block in the list
+     */
+    navigateNextBlock() {
+        if (this.visibleBlocks.length === 0) {
+            return;
+        }
+        
+        this.unhighlightSelectedBlock();
+        
+        if (this.selectedBlockIndex < this.visibleBlocks.length - 1) {
+            this.selectedBlockIndex++;
+        } else {
+            this.selectedBlockIndex = 0;
+        }
+        
+        this.highlightSelectedBlock();
+    }
+
+    /**
+     * Navigate to the previous block in the list
+     */
+    navigatePrevBlock() {
+        if (this.visibleBlocks.length === 0) {
+            return;
+        }
+        
+        this.unhighlightSelectedBlock();
+        
+        if (this.selectedBlockIndex > 0) {
+            this.selectedBlockIndex--;
+        } else {
+            this.selectedBlockIndex = this.visibleBlocks.length - 1;
+        }
+        
+        this.highlightSelectedBlock();
+    }
+
+    /**
+     * Highlight the currently selected block
+     */
+    highlightSelectedBlock() {
+        if (this.selectedBlockIndex === -1 || this.selectedBlockIndex >= this.visibleBlocks.length) {
+            return;
+        }
+        
+        const blockIndex = this.visibleBlocks[this.selectedBlockIndex];
+        const block = this.blocks.blockList[blockIndex];
+        
+        if (!block || !block.container) {
+            return;
+        }
+        
+        // Save the current block for reference
+        this.lastSelectedBlock = block;
+        
+        // Add visual highlight
+        block.container.addChild(this.createHighlightBox(block));
+        
+        // Move block to foreground
+        this.blocks.raiseStackToTop(blockIndex);
+        
+        // Update the blocks.activeBlock
+        this.blocks.activeBlock = blockIndex;
+        
+        // Force a canvas update
+        this.activity.refreshCanvas();
+    }
+
+    /**
+     * Create a highlight box around the selected block
+     * @param {object} block - The block to highlight
+     * @returns {object} - The highlight shape
+     */
+    createHighlightBox(block) {
```

Comment:
We have a block highlight mechanism in block.js. Is there a reason to make a new one?